### ISSUE TITLE
Support metadata `mode`, `uid`, `gid`, `atime`, `mtime` for `rclone mount`

### ISSUE THREAD

**Issue by qarl** [rclone/rclone#7311]
#### What problem are you are trying to solve?

I'd like the virtual filesystem `rclone mount` to support the metadata `mode`, `uid`, `gid`, `atime`, `mtime` in the same way as SSHFS.

#### How do you think rclone should be changed to solve that?
Enable reading and writing of metadata via the virtual filesystem.

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by animosity22** [comment]
Cloud remotes generally don't support that so it's a super edge case.

**Comment by qarl** [comment]
Yes, I understand.  That's why in another issue I'm asking that SFTP support them as well.

The goal here is to replicate the behavior of SSHFS.  With these features, rclone becomes a drop-in replacement with many more advanced features.

**Comment by animosity22** [comment]
Right, very edge so it'll most likely sit unless you sponsor, implement it and/or develop it. Not being mean or anything, but just realistic.

**Comment by qarl** [comment]
I'm posting this because Nick asked me to.

**Comment by animosity22** [comment]
Again, not being mean/sarcastic/etc as Nick is the nicest human on the planet and the backlog is gigantic. If we had 10 of him, life would be great :)

**Comment by mlkt** [comment]
I think we can use additional files to save metadata and then read it during mounting.

**Comment by luxifr** [comment]
> Cloud remotes generally don't support that so it's a super edge case.

for some cloud remotes `rclone` already implements mtime tracking on its own though, eg. through use of tags on the backend object... for instance with azure blob storage, `rclone` stores this in an `mtime` attribute on the object and by default uses this during syncs to determine what to update... 

if there was a bounty program for feature requests, I'd gladly throw a couple of bucks at this, as would some others, too, I'd think

**Comment by qarl** [comment]
At the time I wrote this, I was offering a very sizable bounty for this feature.

**Comment by ncw** [comment]
I think this is a nice feature. With the [metadata framework](https://rclone.org/docs/#metadata) (which quite a few backends support) adjusting the VFS to read/write the metadata for `mode`, `uid`, `git` etc is probably not too difficult.

The metadata supported by the local backend would be a start, with possibly adding xattrs for the remaining metadata.

| Name | Help | Type | Example |
|------|------|------|---------|
| atime | Time of last access | RFC 3339 | 2006-01-02T15:04:05.999999999Z07:00 | 
| btime | Time of file birth (creation) | RFC 3339 | 2006-01-02T15:04:05.999999999Z07:00 | 
| gid | Group ID of owner | decimal number | 500 | 
| mode | File type and mode | octal, unix style | 0100664 | 
| mtime | Time of last modification | RFC 3339 | 2006-01-02T15:04:05.999999999Z07:00 | 
| rdev | Device ID (if special file) | hexadecimal | 1abc | 
| uid | User ID of owner | decimal number | 500 | 

If you wanted this to work with SFTP then we'd need to add metadata support to the sftp backend too.

If you are working on behalf of a company we can do custom development for you as part of a [support contract](https://rclone.com/support) which gets you what you want and keeps the rclone project sustainable.

**Comment by luxifr** [comment]
well, in my case this is for personal use only but I'd already be happy with backends supporting the metadata framework being able to do this (I'm using azure blob storage for storing the mass data part of my Immich setup :))
